src/main/java/com/dreamhorizon/enchantments/enchantments/LifeLeech.java

```diff
@@ -77,9 +77,9 @@ public static void addLifeLeechEffect(ItemStack attackItem, EntityDamageByEntity
         double healthToSetTo = healthGained + playersCurrentHealth;
         
         if (healthToSetTo > healthAttribute.getDefaultValue()) {
-            damager.setHealth(healthAttribute.getValue());
+            damager.setHealth(healthAttribute.getDefaultValue());
```

Comment:
we should be using healthAttribute.getValue() for both the if and the set. If we create armour that increases your health from 20 hearts to 24 in this case it would only heal until 20 hearts. if we use getValue() it heals upto 24 hearts.